internal/controllers/services_controller.go

```diff
@@ -143,8 +103,19 @@ func (s *serviceReconciler) setupController(mgr ctrl.Manager) error {
 }
 
 func createMetricsService(ctx context.Context, deployment *apps.Deployment, apiClient client.Client) error {
+	service := resources.SspMetricsService(deployment.Namespace)
+
+	labels := map[string]string{
+		common.AppKubernetesManagedByLabel: ServiceManagedByLabelValue,
+		common.AppKubernetesVersionLabel:   env.GetOperatorVersion(),
+		common.AppKubernetesComponentLabel: ServiceControllerName,
+	}
 	appKubernetesPartOfValue := deployment.GetLabels()[common.AppKubernetesPartOfLabel]
-	service := ServiceObject(deployment.Namespace, appKubernetesPartOfValue)
+	if appKubernetesPartOfValue != "" {
+		labels[common.AppKubernetesPartOfLabel] = appKubernetesPartOfValue
+	}
+	common.UpdateStringMap(labels, service.GetLabels())
+
```

Comment:
The labels can be simply added to the `service`, without creating a new map or calling `common.UpdateStringMap()`: 
```go
service := resources.SspMetricsService(deployment.Namespace)
service.Labels[common.AppKubernetesManagedByLabel] = ServiceManagedByLabelValue
service.Labels[common.AppKubernetesVersionLabel] = env.GetOperatorVersion()
...
```